module-cellular/modem/ATCommon.cpp

```diff
@@ -156,3 +162,108 @@ Result Channel::checkResult(bsp::cellular::CellularResultCode cellularResult)
 
     return result;
 }
+
+std::string Channel::anonymizeIfNecessary(std::string text)
```

Comment:
IMO this shouldn't be here. Why `Channel` class should be responsible for data anonymization? This should be logger's responsibility - that way not only will it be much clearer from the code logic point of view, but also easier to implement any data anonymization outside `Channel` class (inside the class too, as now every time when you want to add some sensitive logs you need to remember to use these functions instead of just writing the log).

Reply:
Or we can just use LOG_SENSITIVE and adjust it.
AFAIK the LOG_SENSITIVE is doing nothing special, the flag is always ON and if logs exactly the way we pass in the argument.